x/blob/ante/blob_share_decorator_test.go

```diff
@@ -24,141 +30,138 @@ const (
+			blobsPerPFB: 1,
+			blobSize:    100,
+			appVersion:  v2.Version,
 		},
 		{
-			name: "PFB with 1 blob that occupies total square - 1",
-			pfb: &blob.MsgPayForBlobs{
-				BlobSizes: []uint32{uint32(share.AvailableBytesFromSparseShares((squareSize * squareSize) - 1))},
-			},
-			appVersion: v2.Version,
+			name:        "PFB with 1 blob that occupies total square - 1",
+			blobsPerPFB: 1,
+			blobSize:    share.AvailableBytesFromSparseShares(squareSize*squareSize - 1),
+			appVersion:  v2.Version,
+			wantErr:     blob.ErrBlobsTooLarge,
 		},
 		{
-			name: "PFB with 1 blob that occupies total square",
-			pfb: &blob.MsgPayForBlobs{
-				BlobSizes: []uint32{uint32(share.AvailableBytesFromSparseShares(squareSize * squareSize))},
-			},
-			appVersion: v2.Version,
+			name:        "PFB with 1 blob that occupies total square",
+			blobsPerPFB: 1,
+			blobSize:    share.AvailableBytesFromSparseShares(squareSize * squareSize),
+			appVersion:  v2.Version,
 			// This test case should return an error because if the blob
 			// occupies the total square, there is no space for the PFB tx
 			// share.
 			wantErr: blob.ErrBlobsTooLarge,
 		},
 		{
-			name: "PFB with 2 blobs that are 1 share each",
-			pfb: &blob.MsgPayForBlobs{
-				BlobSizes: []uint32{
-					uint32(share.AvailableBytesFromSparseShares(1)),
-					uint32(share.AvailableBytesFromSparseShares(1)),
-				},
-			},
-			appVersion: v2.Version,
+			name:        "PFB with 2 blobs that are 1 share each",
+			blobsPerPFB: 2,
+			blobSize:    100,
+			appVersion:  v2.Version,
 		},
 		{
-			name: "PFB with 2 blobs that occupy half the square each",
-			pfb: &blob.MsgPayForBlobs{
-				BlobSizes: []uint32{
-					uint32(share.AvailableBytesFromSparseShares(squareSize * squareSize / 2)),
-					uint32(share.AvailableBytesFromSparseShares(squareSize * squareSize / 2)),
-				},
-			},
-			appVersion: v2.Version,
-			wantErr:    blob.ErrBlobsTooLarge,
+			name:        "PFB with 2 blobs that occupy half the square each",
+			blobsPerPFB: 2,
+			blobSize:    share.AvailableBytesFromSparseShares(squareSize * squareSize / 2),
+			appVersion:  v2.Version,
+			wantErr:     blob.ErrBlobsTooLarge,
 		},
 	}
 
-	txConfig := encoding.MakeConfig(app.ModuleEncodingRegisters...).TxConfig
+	ecfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			txBuilder := txConfig.NewTxBuilder()
-			require.NoError(t, txBuilder.SetMsgs(tc.pfb))
-			tx := txBuilder.GetTx()
+			kr, _ := testnode.NewKeyring(testfactory.TestAccName)
+			signer, err := user.NewSigner(
+				kr,
+				ecfg.TxConfig,
+				testfactory.ChainID,
+				tc.appVersion,
+				user.NewAccount(testfactory.TestAccName, 1, 0),
+			)
+			require.NoError(t, err)
+
+			blobTx := blobfactory.RandBlobTxs(signer, rand, 1, tc.blobsPerPFB, tc.blobSize)
+
+			btx, isBlob, err := blobtx.UnmarshalBlobTx([]byte(blobTx[0]))
+			require.NoError(t, err)
+			require.True(t, isBlob)
```

Comment:
made this a much more meaningful test by using actual signed txs instead of mocking a pfb size